modules/build-macros/src/test/scala/scala/build/CPSTest.scala

```diff
@@ -50,6 +53,14 @@ class CPSTest extends munit.FunSuite {
     new V
   }
 
+  // When changing something in EitherCPS.scala uncomment this and make sure it does not compile
+  // def ee3: Either[E2, V] = either {
```

Comment:
In Scala 2, we could use [`shapeless.test.illTyped`](https://github.com/milessabin/shapeless/blob/19a9c3a5951e4197084285bd568c2fef6717ef81/core/src/main/scala/shapeless/test/typechecking.scala#L33) to test that this doesn't compile.

Reply:
The problem is that it compiles on Scala 2

Reply:
Oh, ok.